### PR TITLE
fix(components): Refactor DataListSort to use Combobox

### PR DESCRIPTION
**Pull request by rodrigoeidelvein**
## Motivations

Use Combobox instead of Popover on DataList sorting. Also fixes the position of Combobox when it's on the right side of the screen. Before it was overflowing the screen, not it's being adjusted.

### Added

- 

### Changed

- `DataListSort` now uses `Combobox` instead of `Popover`

### Deprecated

- 

### Removed

- 

### Fixed

- `Combobox` overflowing right side of the screen.

### Security

- 

## Testing

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

**Comment by chris-at-jobber** [comment]
Empty / filled: 
image

Note the use of `heading` for the "Sort" in both states

**Comment by chris-at-jobber** [comment]
"Sorted by" as you have it now also works

**Review by chris-at-jobber — COMMENTED** [review]
Design approved! Looks nice, works nice 😄

File: packages/components/src/DataList/components/DataListFilters/components/DataListSort/DataListSort.tsx [hunk 1]
```diff
@@ -1,53 +1,41 @@
...
-      <span ref={divRef}>
-        <Button
-          {...(!state && { icon: "add", iconOnRight: true })}
+    <Combobox
+      onSelect={selection => {
+        handleKeyChange(selection[0].id.toString());
+      }}
+      selected={[{ id: state?.key || "", label: state?.order || "" }]}
+    >
+      <Combobox.Activator>
```
Comment (darryltec) [inline packages/components/src/DataList/components/DataListFilters/components/DataListSort/DataListSort.tsx:22]: nice use of the activator!

File: packages/components/src/DataList/components/DataListFilters/components/DataListSort/DataListSort.tsx [hunk 2]
```diff
@@ -56,7 +44,14 @@ export function DataListSort() {
         const label = headers[sort];
         if (!label) return acc;
 
-        acc.push({ label, value: sort.toString() });
+        if (label === "Last activity") {
+          acc.push({ label: "Last Activity (Newest)", value: `${sort},desc` });
+          acc.push({ label: "Last Activity (Oldest)", value: `${sort},asc` });
```
Comment (darryltec) [inline packages/components/src/DataList/components/DataListFilters/components/DataListSort/DataListSort.tsx:49]: not particularly in love with this. Would be good to add a ticket to allow sorting to replace asc/desc labels per header

Comment (rodrigoeidelvein) [reply packages/components/src/DataList/components/DataListFilters/components/DataListSort/DataListSort.tsx:49]: Me neither. I'll add it

File: docs/components/DataList/Web.stories.tsx
Comment (darryltec) [inline docs/components/DataList/Web.stories.tsx:1]: gotta update the layout too or else this that last column disappears

image

**Review by darryltec — APPROVED** [review]
:shipit:

**Pull request merged**